### ISSUE TITLE
provide API for displaying color blocks like in CSS

### ISSUE THREAD

**Issue by WebFreak001** [Microsoft/vscode#3981]
It would be nice if extensions could tell vscode to put those CSS color blocks in custom file formats.

For example I have a function returning all ranges of CSS color codes in my document and just passing that into some function would be all it needs to do.

Making this a proper API would also make it easier if changes are added to the color block like a future option to make it round or a color picker (https://github.com/Microsoft/vscode/issues/625).

Basically this could be exported for plugin use: https://github.com/Microsoft/vscode/blob/675317b6b856caa2ccb24ce433306a0983116609/src/vs/editor/contrib/color/browser/color.ts#L146-L156

Right now using the TextDecoration API is really hacky because I need to keep track of used colors, remove them and store every different color as separate TextDecoration object which becomes quite hard to handle.

**Comment by jrieken** [comment]
@aeschli @alexandrudima We just talked about this

**Comment by WebFreak001** [comment]
This got added in the latest update
